### ISSUE TITLE
Preserve directory structure, ignore tests, etc. refactoring for ethers-contract-abigen

### ISSUE THREAD

**Issue by meetmangukiya** [gakonst/ethers-rs#1299]
**Is your feature request related to a problem? Please describe.**

Currently, it seems like we generate bindings either as multiple files or single file depending on args, if single file, everything is written to a single file, if multiple file, we create one file per contract and include those in `lib.rs` or `mod.rs` depending on what the user selected, to generate a crate or a module.

Tried this out recently, but see a few problems(?):
1. Everything is generated at top level `src/..contract.rs`, not quite nice.
2. Everything in artifacts dir gets generated, usually we'd want to ignore the contracts defined in `:tests/` or `:src/tests`.

**Describe the solution you'd like**

1. `src/...` contracts get generated in same directory structure as in source and follows same module structure.
2. `lib/...` contracts get generated the same way as in 1, but under a reserved dir like `vendor` / `third_party` directory

**Describe alternatives you've considered**

**Additional context**

**Comment by mattsse** [comment]
> Everything in artifacts dir gets generated, usually we'd want to ignore the contracts defined in :tests/ or :src/tests.

right, ideally we want to support filters, and be default tests should be excluded
